frontend/src/components/Modals/CoffeeChatDetailsModal.tsx

```diff
@@ -18,27 +20,39 @@ const CoffeeChatModal: React.FC<Props> = ({
   open,
   setOpen,
   deleteCoffeeChatRequest,
-  userInfo
+  userInfo,
+  submittedChats
 }) => {
+  const [suggestions, setSuggestions] = useState<CoffeeChatSuggestions>();
   const [membersInCategory, setMembersInCategory] = useState<MemberDetails[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(false);
 
   useEffect(() => {
     const fetchMembers = async () => {
       setIsLoading(true);
-      let suggestions: CoffeeChatSuggestions = {};
       try {
-        suggestions = await CoffeeChatAPI.getCoffeeChatSuggestions(userInfo.email);
+        setSuggestions(await CoffeeChatAPI.getCoffeeChatSuggestions(userInfo.email));
       } catch (error) {
-        suggestions = {};
+        Emitters.generalError.emit({
+          headerMsg: 'Something went wrong',
+          contentMsg: 'Could not retrieve coffee chat suggestions.'
+        });
       }
-      // const categoryMap = new Map<string, MemberDetails[]>(Object.entries(result || {}));
-      setMembersInCategory(suggestions[category] || []);
       setIsLoading(false);
     };
 
     fetchMembers();
-  }, [category, userInfo.email]);
+  }, [userInfo.email]);
+
+  useEffect(() => {
+    setMembersInCategory(
+      suggestions && category in suggestions
```

Comment:
```suggestion
        if (!suggestions || !category) return;
```
You can change this for an early return and then setMembersInCategory. (Just a suggestion!)